client/kubernetes_client_provider.go

```diff
@@ -13,51 +13,92 @@
 package client
 
 import (
+	"errors"
+
+	"github.com/eclipse/che-machine-exec/api/model"
+	"github.com/eclipse/che-machine-exec/cfg"
+	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
-// Provider for creation new kubernetes api client.
-type KubernetesClientProvider struct {
+// K8sAPI object to access k8s api.
+type K8sAPI struct {
 	config *rest.Config
 	client *kubernetes.Clientset
 }
 
-// Create new kubernetes api client provider.
-func NewKubernetesClientProvider() *KubernetesClientProvider {
-	config := createConfig()
-	return &KubernetesClientProvider{
-		config: config,
-		client: createClient(config),
+// NewK8sAPI constructor for creation new k8s api access object.
+func NewK8sAPI(config *rest.Config, client *kubernetes.Clientset) *K8sAPI {
+	return &K8sAPI{config: config, client: client}
+}
+
+// GetClient returns k8s client.
+func (api *K8sAPI) GetClient() *kubernetes.Clientset {
+	return api.client
+}
+
+// GetConfig returns k8s config.
+func (api *K8sAPI) GetConfig() *rest.Config {
+	return api.config
+}
+
+// K8sAPIProvider for creation new K8sAPI.
+type K8sAPIProvider struct {
+	k8sAPI *K8sAPI
+}
+
+// NewK8sAPIProvider creates new K8sAPI provider.
+func NewK8sAPIProvider() *K8sAPIProvider {
+	return &K8sAPIProvider{}
+}
+
+// getK8sAPIWithCA returns k8sApi using service account permissions.
+func (clientProvider *K8sAPIProvider) getK8sAPIWithCA() (*K8sAPI, error) {
+	var err error
```

Comment:
Why do you need it? You should return client OR error, but not both. Please consider getting rid of it.
```suggestion
```